### PR TITLE
Update build, test, artifact process

### PR DESCRIPTION
**Pull request by darinkrauss**
@jh-bate

### CODE REVIEW

File: .travis.yml [hunk 1]
```diff
@@ -1,31 +1,37 @@
 sudo: false
+
 language: node_js
 
 node_js:
-  - "6.10.2"
-  - "stable"
+  - 6.10.2
+  - node
```
Comment (jh-bate) [inline .travis.yml]: what is just the `node` tag mean??

File: .travis.yml [hunk 2]
```diff
@@ -1,31 +1,37 @@
...
 
 cache: yarn
 
 before_install:
-  - if [[ `npm -v` != 4* ]]; then npm i -g npm@4; fi
+  - if [[ `npm -v` != 4* ]]; then npm install -g npm@4; fi
```
Comment (jh-bate) [inline .travis.yml]: just  a sanity check - its intended we use `npm` for the build and not `yarn`??

File: .travis.yml (hunk 1 above)
Comment (darinkrauss) [inline .travis.yml]: According to https://docs.travis-ci.com/user/languages/javascript-with-nodejs/, `node` is the "latest stable Node.js release". The same as `tip`, I believe, but it is explicitly *not* io.js.

File: .travis.yml (hunk 2 above)
Comment (darinkrauss) [inline .travis.yml]: I didn't write this. Jana was the one who updated to `yarn`, but only some of the stuff in the `.travis.yml` and `build.sh` got updated. I don't know the reason behind this, but I am just maintaining the status-quo. If someone else wants to look into whatever the "right thing" is, that'd be great, but it wouldn't be part of this change.

File: .travis.yml (hunk 1 above)
Comment (jh-bate) [inline .travis.yml]: ok

File: .travis.yml (hunk 2 above)
Comment (jh-bate) [inline .travis.yml]: yeah I had a recollection of that just wanted to check

**Pull request merged**